### ISSUE TITLE
Zero initialization of convolutions

### ISSUE THREAD

**Issue by nicolas-dufour** [NVlabs/edm#3]
Hi,
I have observed that the code carefully initialize certain convolutions with zeros init.
Do you have any reference for this kind of design decision?

Thanks!

**Comment by FutureXiang** [comment]
Hi, I am also confused about the weight initialization in different implementations.

## Each implementation has its own initialization style

In the [official DDPM repo](https://github.com/hojonathanho/diffusion), the convs before residual connections and the final conv are initialized with [zeros](https://github.com/hojonathanho/diffusion/blob/1e0dceb3b3495bbe19116a5e1b3596cd0706c543/diffusion_tf/models/unet.py#L53), while other convs are initialized with [zero-mean uniform](https://github.com/hojonathanho/diffusion/blob/1e0dceb3b3495bbe19116a5e1b3596cd0706c543/diffusion_tf/nn.py#L11) distributions.
In the [ADM guided-diffusion repo](https://github.com/openai/guided-diffusion), the convs before residual connections and the final conv are also initialized with [zeros](https://github.com/openai/guided-diffusion/blob/22e0df8183507e13a7813f8d38d51b072ca1e67c/guided_diffusion/unet.py#L210), while others are initialized by PyTorch default.
In the [Score-Based SDE repo](https://github.com/yang-song/score_sde_pytorch/blob/main/models/layers.py), the implementation covers both DDPM/NCSN style initialization.
In this repo, I think it's similar to the Score-Based SDE, but it's still different to the three codebase mentioned above.

## My experiments and observations

Recently, I tried to train diffusion models (DDPM, DDIM, EDM, ...) with the original basic UNet (35.7M #params) on CIFAR-10. Here are some observations:
- I can successfully reproduce the FIDs reported by DDPM and DDIM **without** any custom weight initialization. All parameters are initialized by PyTorch default.
- However, my optimal learning rate **differs** from those in the official repo (`1e-4` vs `2e-4`). When I tried the official one (`2e-4`), the FID result got far worse.
- I train the EDM model with my no-initialization 35.7M mini network with my learning rate, and the results are **reasonable** (better than DDIM).
- However, when I train with the EDM proposed `10e-4` learning rate, the FID result got far worse. To confirm it, I replace the [networks.py](https://github.com/NVlabs/edm/blob/main/training/networks.py) with [mine](https://github.com/FutureXiang/Diffusion/blob/master/model/unet.py) and run with the official EDM code, the FID is still bad.

Seemingly, the mathematical diffusion model (training + sampler) can be decoupled as an individual component. But the neural network model (and its initialization) may be strongly coupled with the hyper-parameters (?).

I wonder if it is really the case, and why the initialization / hyper-parameter matters a lot.
